### DISCUSSION TITLE
How to get Il2Cpp.String into the same format after changing values.

### DISCUSSION THREAD

**Discussion by marioAI** [vfsfitvnm/frida-il2cpp-bridge#234] [Q&A]
Hey!

I'm playing around with fiddling with some function arguments. 
In this case I'm trying to "invoke" the function, but the format seem to be a bit off 

the code:
```
requestinfohandler.implementation = function(this: Il2Cpp.Class | Il2Cpp.Object, queue: Il2Cpp.Object, data: Il2Cpp.String)  {
    console.log("here comes the original data");  
    console.log(data);
    console.log("here comes the changed data");
    const dataNew = Il2Cpp.String.from(data.toString());
    console.log(dataNew);
    (this as Il2Cpp.Class).method("<sendQueue>b__1").invoke(dataNew);
  };
```

The console:
```
here comes the original data
{"group_0_body":{"response":[true]}}
here comes the changed data
"{"group_0_body":{"response":[true]}}"
```

So, how do I get rid of the "" in the dataNew. I think then everything should work.
(maybe thats easy to answer, all I know about JS is what I've learned by doing that example here ;-)

**Comment by bluewave41** [comment]
I've always just done `const dataNew = Il2Cpp.String.from(data.toString().slice(1, -1));` to cut the start and end quotation marks off.

**Reply by marioAI** [reply]
hmm, that just cuts the first and last **{ / }**

console:
```
here comes the original data
{"group_0_body":{"response":{"messages":[],"users":[],"chatStatus":"available"}}}
here comes the changed data
""group_0_body":{"response":{"messages":[],"users":[],"chatStatus":"available"}}"
```

why does the original Il2Cpp.String not have the usual **String - "**

**Reply by bluewave41** [reply]
Actually hang on it shouldn't even matter.

Does the call you have above not work? If you want the raw string you could just do `const raw = data.toString().slice(1, -1);` and then modify that in any way you want.

Once you're done with it you can just send it off with `(this as Il2Cpp.Class).method("<sendQueue>b__1").invoke(Il2Cpp.String.from(raw));` and the extra quotation marks won't affect anything.

**Reply by marioAI** [reply]
Hey and thanks again!

You are absoluetly right. The extra " are only added by the console output.

The real problem was, that the second arg was a Il2Cpp.Object so calling that function with a Il2Cpp.String did not work. ;-)
I just assumed it was a string, because it read so well as .toString() ;-)
